File: SiHyeon/week1/추억점수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    //photo배열에서 모든 값을 꺼낸다
    for (auto page : photo)
    {
        int score = 0; //사진별 그리움 점수
        for (auto people : page)
        {
            //people이름을 name벡터에서 찾아본다.
            auto index = find(name.begin(), name.end(), people);
            if (index != name.end()) //존재하면 인덱스 구해서 더함
                score += yearning[index - name.begin()];
        }
        answer.push_back(score);
    }
    return answer;
}

File: hyeyoung/week1/추억점수.java
package hyeyoung.week1;

import java.util.HashMap;

public class 추억점수 {
    public static void main(String[] args){
        Solution solutions = new Solution();

        String names[] = {"may", "kein", "kain", "radi"};
        int[] yearnings = {5, 10, 1, 3};
        String[][] photos = {{"may", "kein", "kain", "radi"}, {"may", "kein", "brin", "deny"}, {"may", "kein", "brin", "deny"}};

        int[] result = solutions.solution(names, yearnings, photos);

        for (int score : result) {
            System.out.println("Score: " + score);
        }
    }

}

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        HashMap<String, Integer> map1 = new HashMap<>();
        int[] answer = new int[photo.length];

        for(int i = 0; i < name.length; i++){
            map1.put(name[i], yearning[i]);
        }

        for(int i = 0 ; i < photo.length; i++){
            for(int j = 0; j < photo[i].length; j++){
                if(map1.containsKey(photo[i][j]))
                    answer[i] += map1.get(photo[i][j]);
            }
        }

        return answer;
    }
}